### PR TITLE
feat: itinerary maker initial frontend

### PR DESCRIPTION
**Pull request by Arukuen**
This PR features:
- Created a sample component: Dropdown component that can be use in all pages following Flowbite design
- Created an initial Itinerary maker mock up design
    - This proposes separate view for the different stages of itinerary generation

Note that the design is just a proposal (not yet reactive, due to limited time this week) and will implemented when approved.

### CODE REVIEW

**Comment by Arukuen** [comment]
@Qubits-01 My Vercel account uses a different email: sarmientoalquen@gmail.com, which might be causing the error.

**Comment by nidec-ct-dev** [comment]
> @Qubits-01 My Vercel account uses a different email: [sarmientoalquen@gmail.com](mailto:sarmientoalquen@gmail.com), which might be causing the error.

Nope. It still uses the Nidec's Vercel account. Also, you can click the *inspect* link to view the deployment logs and KNOW what specific errors are causing the build to fail.

Do take note that before deploying or submitting a PR, PLEASE RUN ```npm run lint``` FIRST IN YOUR LOCAL DEV ENVIRONMENT TO AVOID TRIVIAL ERRORS SUCH AS THESE. I configured our project's eslint for this very reason. Thanks!

**Comment by Qubits-01** [comment]
> @Qubits-01 My Vercel account uses a different email: [sarmientoalquen@gmail.com](mailto:sarmientoalquen@gmail.com), which might be causing the error.

Nope. It still uses Nidec's Vercel account. Also, you can click the *inspect* link to view the deployment logs and KNOW what specific errors are causing the build to fail.
image

Do take note that before deploying or submitting a PR, PLEASE RUN ```npm run lint``` FIRST IN YOUR LOCAL DEV ENVIRONMENT TO AVOID TRIVIAL ERRORS SUCH AS THESE. I configured our project's eslint for this very reason. Thanks!

File: .vscode/settings.json
Comment (Qubits-01) [inline .vscode/settings.json:1]: NO. Don't change this. VS code's default typescript formatter is not sufficient. That is why our project is using prettier (and there are attached custom configurations to it unique to this project).

File: app/components/dropdown.tsx
Comment (Qubits-01) [inline app/components/dropdown.tsx:1]: The general components folder should be in the root dir.

File: app/dashboard/itinerary-maker/page.tsx [hunk 1]
```diff
@@ -1,7 +1,255 @@
-import React from "react";
+'use client';
+
+import React, { useState } from "react";
```
Comment (Qubits-01) [inline app/dashboard/itinerary-maker/page.tsx:3]: Don't use useState if you will not use it.

**Review by Qubits-01 — CHANGES_REQUESTED** [review]

File: components/dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
+"use client";
+
+import React, { FC, useState } from 'react';
+
+interface DropdownProps {
```
Comment (Qubits-01) [inline components/dropdown.tsx:5]: You do not need to manually define your types. Flowbite has built-in types for these kinds of things. Please read the docs.
image

**Review by Qubits-01 — CHANGES_REQUESTED** [review]

**Comment by Qubits-01** [comment]
Good job though, your Vercel build is working ✅

**Pull request merged**